src/reconcile/rec_write.c

```diff
@@ -638,6 +646,11 @@ __rec_cleanup(WT_SESSION_IMPL *session, WT_RECONCILE *r)
 
     /* Reconciliation is not re-entrant, make sure that doesn't happen. */
     r->ref = NULL;
+
+    /* Clear stats related data. */
+    r->page_with_ts = false;
```

Comment:
We do not need to clear things at the end, instead we need to initialize these in __rec_init(). See r->cache_write_hs as an example.

Reply:
Fixed